linera-views/src/common.rs

```diff
@@ -190,57 +256,228 @@ pub trait KeyValueStoreClient {
         Ok(Some(big_value))
     }
 
-    /// Reads a single `key` and deserializes the result if present.
-    async fn read_key_big<V: DeserializeOwned>(&self, key: &[u8]) -> Result<Option<V>, Self::Error>
-    where
-        Self::Error: From<bcs::Error> + From<DatabaseConsistencyError>,
-    {
-        from_bytes_opt(self.read_key_big_bytes(key).await?)
-    }
-
-    /// Retrieves multiple `Vec<u8>` from the database using the provided `keys`.
     async fn read_multi_key_bytes(
         &self,
         keys: Vec<Vec<u8>>,
-    ) -> Result<Vec<Option<Vec<u8>>>, Self::Error>;
+    ) -> Result<Vec<Option<Vec<u8>>>, Self::Error> {
+        let mut big_keys = Vec::new();
+        for key in &keys {
+            let mut big_key = key.clone();
+            big_key.extend(&[0]);
+            big_keys.push(big_key);
+        }
+        let values = self.client.read_multi_key_bytes(big_keys).await?;
+        let mut big_values = Vec::<Option<Vec<u8>>>::new();
+        let mut keys_add = Vec::new();
+        let mut n_blocks = Vec::new();
+        for (key, value) in keys.iter().zip(values) {
```

Comment:
I think we almost always zip the result with the keys after reading multiple values. Since `read_multi_key_bytes` takes ownership of the keys anyway, we could return the key-value pairs instead of just the values, so we don't need to zip afterwards?

(In this case it will be big keys, of course, but we could just drop the `0` again where needed?)